pkg/homedir/homedir_unix.go

```diff
@@ -133,6 +134,33 @@ func GetConfigHome() (string, error) {
 	return rootlessConfigHomeDir, rootlessConfigHomeDirError
 }
 
+// checkBootId checks if the boot_id of the system has changed since the last time the runtime directory was created.
+func checkBootId(path string) error {
+	systemBootId, err := os.ReadFile("/proc/sys/kernel/random/boot_id")
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// The system does not have a boot_id, so ignore the check
+			return nil
+		}
+		return err
+	}
+
+	bootIdFile := filepath.Join(path, "bootid")
+	bootId, err := os.ReadFile(bootIdFile)
+	if err != nil {
+		// If the file does not exist, create it
+		if os.IsNotExist(err) {
+			return os.WriteFile(bootIdFile, systemBootId, 0o600)
```

Comment:
there are no locks for this so in theory there is a super unlikely race between two callers calling this check at the same time.

WriteFile() can cause a empty/partial written file and then the other caller would not read the full content thus failing the check below incorrectly.
So in short I think this must use the atomic write function we have here that uses rename to ensure we have no partial file.